minio/minio
# Allow caching based on a configurable number of hits.

This PR also moves cache locking down to per cache drive

## Description


## Motivation and Context
Allow option to cache only frequently used objects by specifying a minimum number of hits before
an object is cached.


## How to test this PR?
```
 mc admin config set myminio cache  drives=/tmp/cache exclude= expiry=90 quota=99 after=2
mc admin service restart myminio
```
objects cached on GET or single PUT should honor the minimum hits configured in cache settings before caching the object.
## Types of changes

## Checklist:
